WordPress/Sniffs/Theme/NoFaviconSniff.php

```diff
@@ -0,0 +1,76 @@
+<?php
+/**
+ * WordPress Coding Standard.
+ *
+ * @package WPCS\WordPressCodingStandards
+ * @link    https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards
+ * @license https://opensource.org/licenses/MIT MIT
+ */
+
+/**
+ * Check for hardcoded favicons instead of using core implementation.
+ *
+ * @link    https://make.wordpress.org/themes/handbook/review/required/#core-functionality-and-features
+ *
+ * @package WPCS\WordPressCodingStandards
+ *
+ * @since   0.xx.0
+ */
+class WordPress_Sniffs_Theme_NoFaviconSniff implements PHP_CodeSniffer_Sniff {
+
+	const REGEX_TEMPLATE = '` (%s)`i';
+	const REGEX_ATTR_TEMPLATE = '%1$s=[\'"]%2$s[\'"]';
+
+	protected $attribute_blacklist = array(
+		'rel' => array(
+			'shortcut',
+			'icon',
+			'apple-touch-icon',
+			'apple-touch-icon-precomposed',
+		),
+		'name' => array(
+			'msapplication-TileImage'
+		),
+	);
+
+	protected $favicon_regex;
+
+	/**
+	 * Returns an array of tokens this test wants to listen for.
+	 *
+	 * @return array
+	 */
+	public function register() {
+		$regex_parts = array();
+		foreach( $this->attribute_blacklist as $key => $values ) {
+			$values = array_map( 'preg_quote', $values );
```

Comment:
Oh and to be extremely precise, we should pass the delimiter to preg_quote, though it's not going to make any difference in this case:
```php
$values = array_map( 'preg_quote', $values, array_fill ( 0, count( $values ), '`' ) );
```